dialer.go

```diff
@@ -126,8 +129,12 @@ func NewDialer(ctx context.Context, opts ...Option) (*Dialer, error) {
 		if err != nil {
 			return nil, fmt.Errorf("failed to create token source: %v", err)
 		}
-		cfg.tokenSource = ts
 		cfg.sqladminOpts = append(cfg.sqladminOpts, option.WithTokenSource(ts))
+		scoped, err := google.DefaultTokenSource(ctx, loginScope)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create scoped token source: %v", err)
+		}
+		cfg.tokenSource = scoped
```

Comment:
Can we explicitly name this `iamTokenSource` so it's clear we're only using it for Auto IAM AUthN?

Reply:
Definitely. Good idea too.